PRO-16890 : Enforce required flags for provar:metadatacache

MayankNehru opened the pull request:
The command started executing even when required flags were not given. Changed that to give error in typescript itself. Also made changes to make sure that the flags work correctly in tandem.

Commit:
PRO-16890 : Enforce required flags for provar:metadatacache

The command started executing even when required flags were not given.
Changed that to give error in typescript itself. Also made changes to
make sure that the flags work correctly in tandem.

riteshnahar commented on src/commands/provar/metadatacache.ts:
```diff
@@ -81,6 +81,24 @@ export default class MetadataCache extends SfdxCommand {
+            return {};
+        }
+
+        if (
```
Change it to:
`if (!["Reload", "Refresh", "Reuse"].includes(metadataLevel))`

riteshnahar commented on src/commands/provar/metadatacache.ts:
```diff
@@ -117,26 +135,26 @@ export default class MetadataCache extends SfdxCommand {
         }
 
-        const userInfoString = provarDxUtils.prepareRawProperties(
+        const userInfoString = connections && userInfo == null ? "NA" : provarDxUtils.prepareRawProperties(
```
`userInfo === null`

riteshnahar reviewed (changes requested):
Please do necessary changes in `CHANGELOG.md` file as well

Commit:
PRO-16890 : Implement review comments and add change log

Commit:
PRO-16890 : Implement review comments

Corrected conditional checks to use "===" & '!==' for comparison and
renamed variables with self explanatory names

simoncarter-provar commented on src/commands/provar/metadatacache.ts:
```diff
@@ -81,6 +81,22 @@ export default class MetadataCache extends SfdxCommand {
+            return {};
+        }
+
+        (!["Reload", "Refresh", "Reuse"].includes(metadataLevel)) {
```
Should this be an if statement?

`if (!["Reload", "Refresh", "Reuse"].includes(metadataLevel)) {`

Commit:
PRO-16890 : Implement review comments

Commit:
Merge branch 'development' into PRO-16890
